ROX:26971: Add CRS to ACS docs (#89252)
openshift/openshift-docs: modules/acs-quick-install-secured-cluster-using-helm.adoc

* You must have the *Central API Endpoint* address. You can view this information by choosing *Advanced Cluster Security* -> *ACS Instances* from the Red{nbsp}Hat Hybrid Cloud Console navigation menu, then clicking the ACS instance you created.
endif::[]

//Procedure for kubernetes - non-cloud
//do not show for openshift, cloud
ifndef::openshift,cloud-svc[]
.Procedure

> * Run one of the following commands on your Kubernetes based clusters:

Reviewer: ```suggestion
* Run one of the following commands on your Kubernetes-based clusters:
```

> ** If you are using an init bundle, run the following command:

Reviewer 2: It doesn't mention what to do in case the user has decided to try out CRS?
But in the section below it does. I see the problem of duplication...
but this way it suggests that CRSes can only be used on OpenShift Container plus, not on vanilla K8s?

Author: This is what it shows for Kubernetes (non-ocp):

![Screenshot 2025-03-05 at 11 44 57 AM](https://github.com/user-attachments/assets/7800ffcf-ab48-43d4-b904-d88ecf1065ac)

![Screenshot 2025-03-05 at 11 45 13 AM](https://github.com/user-attachments/assets/6d70a8f5-9ebe-4078-a630-bca37d8e406e)

Author: This is what it shows for OpenShift:

![Screenshot 2025-03-05 at 11 47 23 AM](https://github.com/user-attachments/assets/2d267def-5e5a-4047-93f3-f40b8cc2f66f)

![Screenshot 2025-03-05 at 11 47 34 AM](https://github.com/user-attachments/assets/d69d2556-7347-424c-8019-a36db309d6b5)

Author: > It doesn't mention what to do in case the user has decided to try out CRS? But in the section below it does. I see the problem of duplication... but this way it suggests that CRSes can only be used on OpenShift Container plus, not on vanilla K8s?

I pasted screenshots from the preview that show what the conditionals are doing. I think they are correct, but please let me know if not. They are kind of confusing to me when I do them, so I might have set them up incorrectly.

Reviewer 2: Hmm, ok, probably I have misread -- sorry!

Author: It is difficult to parse when just looking at the source, but if it's not showing up correctly in the HTML, please lmk.

    -f <path_to_cluster_init_bundle.yaml> \// <1>
    -f <path_to_pull_secret.yaml> \// <2>
    --set clusterName=<name_of_the_secured_cluster> \

>     --set centralEndpoint=<endpoint_of_central_service>\// <3>

Reviewer 2: Maybe a whitespace missing before the trailing backslash in this line.

Author: Thanks, did a search & replace and I hope fixed all of them.

    --set imagePullSecrets.password=<your redhat.com password>//<5>
----
<1> Use the `-f` option to specify the path for the init bundle.
<2> Use the -f option to specify the path for the pull secret for Red{nbsp}Hat Container Registry authentication.
<3> Specify the address and port number for Central. For example, `acs.domain.com:443`.
<4> Include the user name for your pull secret for Red{nbsp}Hat Container Registry authentication.
<5> Include the password for your pull secret for Red{nbsp}Hat Container Registry authentication.
** If you are using a CRS, run the following command:
+
[source,terminal]
----
$ helm install -n stackrox --create-namespace \
    stackrox-secured-cluster-services rhacs/secured-cluster-services \
    --set-file crs.file=<crs_file_name.yaml> \// <1>
    -f <path_to_pull_secret.yaml> \// <2>
    --set clusterName=<name_of_the_secured_cluster> \
    --set centralEndpoint=<endpoint_of_central_service> \// <3>
    --set imagePullSecrets.username=<your redhat.com username> \// <4>
    --set imagePullSecrets.password=<your redhat.com password>// <5>
----
<1> Use the name of the file in which the generated CRS has been stored.

> <2> Use the -f option to specify the path for the pull secret for Red{nbsp}Hat Container Registry authentication.

Reviewer: ```suggestion
<2> Use the `-f` option to specify the path for the pull secret for Red{nbsp}Hat Container Registry authentication.
```

<3> Specify the address and port number for Central. For example, `acs.domain.com:443`.
<4> Include the user name for your pull secret for Red{nbsp}Hat Container Registry authentication.
<5> Include the password for your pull secret for Red{nbsp}Hat Container Registry authentication.
endif::openshift,cloud-svc[]

//Procedure for kubernetes - cloud
ifdef::cloud-svc[]
.Procedure
* Run one of the following commands on your Kubernetes based clusters:
** If you are using an init bundle, run the following command:
+
[source,terminal]
----
$ helm install -n stackrox --create-namespace \
    stackrox-secured-cluster-services rhacs/secured-cluster-services \
    -f <path_to_cluster_init_bundle.yaml> \// <1>
    -f <path_to_pull_secret.yaml> \// <2>
    --set clusterName=<name_of_the_secured_cluster> \
    --set centralEndpoint=<endpoint_of_central_service> \// <3>
    --set imagePullSecrets.username=<your redhat.com username> \//<4>
    --set imagePullSecrets.password=<your redhat.com password>//<5>
----
<1> Use the `-f` option to specify the path for the init bundle.

> <2> Use the -f option to specify the path for the pull secret for Red{nbsp}Hat Container Registry authentication.

Reviewer: ```suggestion
<2> Use the `-f` option to specify the path for the pull secret for Red{nbsp}Hat Container Registry authentication.
```

<3> Enter the *Central API Endpoint* address. You can view this information by choosing *Advanced Cluster Security* -> *ACS Instances* from the Red{nbsp}Hat Hybrid Cloud Console navigation menu, then clicking the {product-title-short} instance you created.
<4> Include the user name for your pull secret for Red{nbsp}Hat Container Registry authentication.
<5> Include the password for your pull secret for Red{nbsp}Hat Container Registry authentication.
endif::cloud-svc[]

//Procedure for OpenShift - non-cloud
//do not show for cloud
ifndef::cloud-svc[]

.Procedure

> * Run one of the following commands on {ocp} clusters:

Reviewer: ??
```suggestion
* Run one of the following commands on an {ocp} cluster:
```

    -f <path_to_pull_secret.yaml> \// <2>
    --set clusterName=<name_of_the_secured_cluster> \
    --set centralEndpoint=<endpoint_of_central_service> \// <3>
    --set scanner.disable=false <4>
----
<1> Use the `-f` option to specify the path for the init bundle.
<2> Use the -f option to specify the path for the pull secret for Red{nbsp}Hat Container Registry authentication.
<3> Specify the address and port number for Central. For example, `acs.domain.com:443`.
<4> Set the value of the `scanner.disable` parameter to `false`, which means that Scanner-slim will be enabled during the installation. In Kubernetes, the secured cluster services now include Scanner-slim.
** If you are using a CRS, run the following command:
+
[source,terminal]
----
$ helm install -n stackrox --create-namespace \
    stackrox-secured-cluster-services rhacs/secured-cluster-services \
    --set-file crs.file=<crs_file_name.yaml> \// <1>
    -f <path_to_pull_secret.yaml> \// <2>
    --set clusterName=<name_of_the_secured_cluster> \
    --set centralEndpoint=<endpoint_of_central_service> \// <3>
    --set scanner.disable=false <4>
----
<1> Use the name of the file in which the generated CRS has been stored.

> <2> Use the -f option to specify the path for the pull secret for Red{nbsp}Hat Container Registry authentication.

Reviewer: ```suggestion
<2> Use the `-f` option to specify the path for the pull secret for Red{nbsp}Hat Container Registry authentication.
```

<3> Specify the address and port number for Central. For example, `acs.domain.com:443`.
<4> Set the value of the `scanner.disable` parameter to `false`, which means that Scanner-slim will be enabled during the installation. In Kubernetes, the secured cluster services now include Scanner-slim.
endif::cloud-svc[]

//Procedure for OpenShift cloud
ifdef::cloud-svc[]

.Procedure

> * Run one of the following commands on {ocp} clusters:

Reviewer: ??
```suggestion
* Run one of the following commands on an {ocp} cluster:
```

** If you are using an init bundle, run the following command:
+
[source,terminal]
----
$ helm install -n stackrox --create-namespace \
    stackrox-secured-cluster-services rhacs/secured-cluster-services \
    -f <path_to_cluster_init_bundle.yaml> \// <1>
    -f <path_to_pull_secret.yaml> \// <2>
    --set clusterName=<name_of_the_secured_cluster> \
    --set centralEndpoint=<endpoint_of_central_service> \// <3>
    --set scanner.disable=false <4>
----
<1> Use the `-f` option to specify the path for the init bundle.

> <2> Use the -f option to specify the path for the pull secret for Red{nbsp}Hat Container Registry authentication.

Reviewer: ```suggestion
<2> Use the `-f` option to specify the path for the pull secret for Red{nbsp}Hat Container Registry authentication.
```

<3> Enter the *Central API Endpoint* address. You can view this information by choosing *Advanced Cluster Security* -> *ACS Instances* from the Red{nbsp}Hat Hybrid Cloud Console navigation menu, then clicking the {product-title-short} instance you created.

> <4> Set the value of the `scanner.disable` parameter to `false`, which means that Scanner-slim will be enabled during the installation. In Kubernetes, the secured cluster services now include Scanner-slim as an optional component.

Reviewer 2: nit: a priori the scanner components are optional for secured clusters -- they can be installed for enabling the local scanning feature or they can be left out.

But, regarding the wording, if I set `scanner.disable=false`, then this instructs the Helm chart to also *include* the scanner, at which it wouldn't make too much sense (IMHO) to still denote it as being "optional". 
```suggestion
<4> Set the value of the `scanner.disable` parameter to `false`, which means that Scanner-slim will be enabled during the installation. In Kubernetes, the secured cluster services now include Scanner-slim.
```

** If you are using a CRS, run the following command:
+
[source,terminal]
----
$ helm install -n stackrox --create-namespace \
    stackrox-secured-cluster-services rhacs/secured-cluster-services \
    --set-file crs.file=<crs_file_name.yaml> \// <1>
    -f <path_to_pull_secret.yaml> \// <2>
    --set clusterName=<name_of_the_secured_cluster> \
    --set centralEndpoint=<endpoint_of_central_service> \// <3>
    --set scanner.disable=false <4>
----
<1> Use the name of the file in which the generated CRS has been stored.

> <2> Use the -f option to specify the path for the pull secret for Red{nbsp}Hat Container Registry authentication.

Reviewer: ```suggestion
<2> Use the `-f` option to specify the path for the pull secret for Red{nbsp}Hat Container Registry authentication.
```

<3> Enter the *Central API Endpoint* address. You can view this information by choosing *Advanced Cluster Security* -> *ACS Instances* from the Red{nbsp}Hat Hybrid Cloud Console navigation menu, then clicking the {product-title-short} instance you created.

> <4> Set the value of the `scanner.disable` parameter to `false`, which means that Scanner-slim will be enabled during the installation. In Kubernetes, the secured cluster services now include Scanner-slim as an optional component.

Reviewer 2: same here.

> <4> Set the value of the `scanner.disable` parameter to `false`, which means that Scanner-slim will be enabled during the installation. In Kubernetes, the secured cluster services now include Scanner-slim as an optional component.

Reviewer 2: I actually wonder, why is this `scanner.disable=false` setting part of this command (maybe I am overlooking something...)?

Author: I'm not sure - this was prior text, so I just kept it as is.

Author: I cannot figure out when this was added to the docs (it's been there for quite some time) so it's hard to know why that line is there. I can research, but for now, I hope it's OK to just make the change you are recommending and then figure out why it's specifically called out and if we need to keep that language.
